src/mongoClusters/wizards/addWorkspaceConnection/ConnectionStringStep.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzureWizardPromptStep } from '@microsoft/vscode-azext-utils';
+import ConnectionString from 'mongodb-connection-string-url';
+import { localize } from '../../../utils/localize';
+import { type AddWorkspaceConnectionContext } from './AddWorkspaceConnectionContext';
+
+export class ConnectionStringStep extends AzureWizardPromptStep<AddWorkspaceConnectionContext> {
+    public hideStepCount: boolean = true;
+
+    public async prompt(context: AddWorkspaceConnectionContext): Promise<void> {
+        const prompt: string = localize(
+            'mongoClusters.addWorkspaceConnection.connectionString.prompt',
+            'Enter the connection string of your MongoDB cluster.',
+        );
+        context.connectionString = (
+            await context.ui.showInputBox({
+                prompt: prompt,
+                ignoreFocusOut: true,
+                placeHolder: localize(
+                    'mongoClusters.addWorkspaceConnection.connectionString.placeholder',
+                    'Starts with mongodb:// or mongodb+srv://',
+                ),
+                validateInput: ConnectionStringStep.validateInput,
+                asyncValidationTask: (connectionString: string) => this.validateConnectionString(connectionString),
+            })
+        ).trim();
+
+        context.valuesToMask.push(context.connectionString);
```

Comment:
maybe add connectionString.username and connectionString.password to the mask list as single strings as well, just to be on the safe side.

Reply:
good catch, added masking of these in their "steps"